### ISSUE TITLE
Narrator doesn't announce TextPattern elements

### ISSUE THREAD

**Issue by vladimir-krestov** [dotnet/winforms#6020]
* .NET Core Version: 7.0 (main)
* Doesn't repro for .NET 6.0 (release)
* Regression from #5404 

* Have you experienced this same bug with .NET Framework?: No

**Problem description:**
- TextPattern doesn't for for TextBox

**Expected behavior:**
- TextPattern should work as before (.NET 6.0)

**Minimal repro:**
- Run a 7.0 test app with a TextBox
- Run Narrator
- Try to input/select/navigate in the TextBox
